pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -1165,7 +1165,16 @@ class ModelParser implements Parser {
      */
     protected ModelASTValue parseArgument(Expression e) {
         if (e instanceof ConstantExpression) {
-            return ModelASTValue.fromConstant(e.value, e)
+            Object val = e.value
+            if (val instanceof BigDecimal) {
+                val = val.doubleValue()
+            } else if (val instanceof BigInteger) {
+                if (val > Long.MAX_VALUE || val < Long.MIN_VALUE) {
+                    errorCollector.error(ModelASTValue.fromConstant(-1, e), Messages.ModelParser_BigIntegerValue())
+                }
+                val = -1
```

Comment:
Whoops, yup. That `-1` (when it's put in the right place!) ends up being ignored in practice, because the error trumps it, but yeah, it is in the wrong place.

Reply:
Oh, and this situation can only, in practice, be hit if you specify a constant number greater than `Long.MAX_VALUE` - we're only handling `ConstantExpression`s here, so, say, `24` ends up as functionally equivalent to `new Integer(24)`, while `2147483648` (i.e., one more than `Integer.MAX_VALUE`) will be equivalent to `new Long(2147483648)`. Groovy handles that behind the scenes.